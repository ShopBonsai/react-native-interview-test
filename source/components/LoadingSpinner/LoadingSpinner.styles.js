import styled from 'styled-components'
import { View, Text, ActivityIndicator } from 'react-native'

export const LoadingSpinnerContainer = styled(View)`
  flex: 1;
  position: absolute;
  width: 100%;
  height: 100%;
  justify-content: center;
  align-items: center;
`

export const Spinner = styled(ActivityIndicator)`
  color: #fff;
`
export const LoadingText = styled(Text)`
  color: white;
  font-weight: bold;
  padding: 5px;
`
export default {
  LoadingSpinnerContainer
}
