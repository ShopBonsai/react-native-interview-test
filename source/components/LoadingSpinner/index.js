import React from 'react'
import { LoadingSpinnerContainer, Spinner, LoadingText } from './LoadingSpinner.styles'

export default function LoadingSpinner() {
  return (
    <LoadingSpinnerContainer>
      <Spinner size="large" />
      <LoadingText>Loading...</LoadingText>
    </LoadingSpinnerContainer>
  )
}
