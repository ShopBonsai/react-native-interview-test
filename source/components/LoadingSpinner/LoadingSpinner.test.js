// libraries
import React from 'react'
import renderer from 'react-test-renderer'
import LoadingSpinner from './index'


it('Renders LoadingSpinner correctly', () => {
  const tree = renderer
    .create(<LoadingSpinner />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
