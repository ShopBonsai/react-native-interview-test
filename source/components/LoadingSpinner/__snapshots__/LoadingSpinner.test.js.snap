// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders LoadingSpinner correctly 1`] = `
<View
  className="sc-bdVaJa gucMGt"
>
  <ActivityIndicator
    animating={true}
    className="sc-bwzfXH gdcgOT"
    color="#999999"
    hidesWhenStopped={true}
    size="large"
  />
  <Text
    className="sc-htpNat DjMmj"
  >
    Loading...
  </Text>
</View>
`;
